auth.py

```diff
@@ -0,0 +1,104 @@
+import json
+import logging
+
+from base64 import b64decode
+from urllib.parse import unquote
+
+import boto3
+from bcrypt import checkpw
+
+from logger import log
+
+
+class AuthenticationError(Exception):
+    pass
+
+
+class BasicAuthDecodeError(Exception):
+    pass
+
+
+class BasicAuthHandler(object):
+    def __init__(self, config):
+        self.dynamo_region = config["DYNAMODB_REGION"]
+        self.dynamo_client = boto3.client("dynamodb", region_name=self.dynamo_region)
+        self.dynamo_table = config["DYNAMODB_TABLE_NAME"]
+
+    def check_header(self, basicauth_header):
+        """
+        Checks Basic Auth header value against username/password combos
+        in dynamodb.
+
+        Returns True if it finds a match, else it returns False.
+        """
+        try:
+            username, password = self.decode(basicauth_header)
+        except BasicAuthDecodeError:
+            return False
+
+        resp = self.dynamo_client.get_item(
+            TableName=self.dynamo_table, Key={"username": {"S": username}}
```

Comment:
Yeah I'm wondering if the Dynamo table doesn't add complexity considering it's basically going to just be read only and only contain a few items -- one other option I guess too could be we just pass the entire configuration for this function in as a JSON doc in an environment variable, that just get's decrypted via sops when we deploy it

Reply:
I'm good with that. I'll refactor it to grab a json doc stored in S3 that is encrypted using sops. Do you think we should stick with using bcrypt for the passwords?